### PR TITLE
Improve attestation validation

### PR DESCRIPTION
**Pull request by janakamarasena**
- Remove error response in attestation proxy when api based auth is disabled and the proxy is engaged.
- Add attestation check to authz ep

### CODE REVIEW

File: components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java [hunk 1]
```diff
@@ -292,4 +299,12 @@ private Response buildResponse(String errorDescription, Response.Status status)
 
         return Response.status(status).entity(errorJSON).build();
     }
+
+    private static String removeTrailingSlash(String url) {
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java:303]: Do we have any purpose for this method as static

File: components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java [hunk 2]
```diff
@@ -110,18 +111,23 @@ public void handleMessage(Message message) {
...
                                         serviceProvider.getApplicationResourceId(),
                                         IdentityTenantUtil.resolveTenantDomain());
-                        // Set the client attestation context in the HTTP request
-                        setContextToRequest(request, clientAttestationContext);
                     } else {
-                        throw new WebApplicationException(buildResponse("Client is not registered for " +
-                                        "API Based Authentication Service", Response.Status.BAD_REQUEST));
+                        /* In order for client attestation to be enabled it requires API-based authentication to be
+                         enabled. Therefore, if API-based authentication is disabled, client attestation is disabled.*/
+                        clientAttestationContext = new ClientAttestationContext();
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java:125]: With this change, Response mode : direct can be called without enabling API Based Authentication is this expected behaviour?

Comment (janakamarasena) [reply components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java:125]: This check will be done in api based auth service level when UI support is available.

File: components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java (hunk 1 above)
Comment (janakamarasena) [reply components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java]: fixed.

**Review by Thumimku — APPROVED** [review]
LGTM

**Pull request merged**